dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataFlusher.kt

```diff
@@ -9,28 +9,35 @@ package com.datadog.android.core.internal.data.upload
 import androidx.annotation.WorkerThread
 import com.datadog.android.core.internal.net.DataUploader
 import com.datadog.android.core.internal.persistence.PayloadDecoration
-import com.datadog.android.core.internal.persistence.file.ChunkedFileHandler
+import com.datadog.android.core.internal.persistence.file.FileMover
 import com.datadog.android.core.internal.persistence.file.FileOrchestrator
+import com.datadog.android.core.internal.persistence.file.batch.BatchFileReader
+import com.datadog.android.core.internal.persistence.file.existsSafe
 import com.datadog.android.core.internal.utils.join
 
 internal class DataFlusher(
     internal val fileOrchestrator: FileOrchestrator,
     internal val decoration: PayloadDecoration,
-    internal val handler: ChunkedFileHandler
+    internal val fileReader: BatchFileReader,
+    internal val fileMover: FileMover
 ) : Flusher {
 
     @WorkerThread
     override fun flush(uploader: DataUploader) {
         val toUploadFiles = fileOrchestrator.getFlushableFiles()
         toUploadFiles.forEach {
-            val batch = handler.readData(it)
+            val batch = fileReader.readData(it)
                 .join(
                     separator = decoration.separatorBytes,
                     prefix = decoration.prefixBytes,
                     suffix = decoration.suffixBytes
                 )
             uploader.upload(batch)
-            handler.delete(it)
+            fileMover.delete(it)
+            val metaFile = fileOrchestrator.getMetadataFile(it)
```

Comment:
where is this data uploaded if you delete it here ?

Reply:
The data is uploaded few lines above (`uploader.upload(batch)`), basically the logic is the same, I just added metadata deletion.

But I see your concern: metadata is requested for the file which is deleted and while it is ok to have (`File` object stays there still and we just use its path to get associated metadata file), I swapped the lines - now metadata file is requested before batch file deletion, which makes the operation more clear.